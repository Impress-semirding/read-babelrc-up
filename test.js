import path from 'path';
import test from 'ava';
import fn from '.';

const cwd = path.join('fixture', 'app');
const babelrcPath = path.resolve('fixture', '.babelrc');

test('async', async t => {
	const x = await fn({cwd});
	t.is(x.path, babelrcPath);
	t.deepEqual(x.babel.presets, ['es2015', 'react']);
	t.deepEqual(x.babel.plugins, ['transform-class-properties']);
});

test('sync', t => {
	const x = fn.sync({cwd});
	t.is(x.path, babelrcPath);
	t.deepEqual(x.babel.presets, ['es2015', 'react']);
	t.deepEqual(x.babel.plugins, ['transform-class-properties']);
});
