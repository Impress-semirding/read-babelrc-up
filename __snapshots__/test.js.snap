// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async 1`] = `
Object {
  "babel": Object {
    "plugins": Array [
      "transform-class-properties",
    ],
    "presets": Array [
      "es2015",
      "react",
    ],
  },
  "path": "/Users/akameco/src/github.com/akameco/read-babelrc-up/fixture/babelrc/.babelrc",
}
`;

exports[`parse babelrc.js - sync 1`] = `
Object {
  "babel": Object {
    "plugins": Array [
      "transform-class-properties",
    ],
    "presets": Array [
      "es2015",
      "react",
    ],
  },
  "path": "/Users/akameco/src/github.com/akameco/read-babelrc-up/fixture/babelrcjs/.babelrc.js",
}
`;

exports[`parse babelrc.js 1`] = `
Object {
  "babel": Object {
    "plugins": Array [
      "transform-class-properties",
    ],
    "presets": Array [
      "es2015",
      "react",
    ],
  },
  "path": "/Users/akameco/src/github.com/akameco/read-babelrc-up/fixture/babelrcjs/.babelrc.js",
}
`;

exports[`parse package.json - sync 1`] = `
Object {
  "babel": Object {
    "plugins": Array [
      "transform-class-properties",
    ],
  },
  "path": "/Users/akameco/src/github.com/akameco/read-babelrc-up/fixture/pkg/package.json",
}
`;

exports[`parse package.json - sync 2`] = `
Object {
  "babel": Object {
    "plugins": Array [
      "transform-class-properties",
    ],
  },
  "path": "/Users/akameco/src/github.com/akameco/read-babelrc-up/fixture/pkg/package.json",
}
`;

exports[`parse package.json 1`] = `
Object {
  "babel": Object {
    "plugins": Array [
      "transform-class-properties",
    ],
  },
  "path": "/Users/akameco/src/github.com/akameco/read-babelrc-up/fixture/pkg/package.json",
}
`;

exports[`sync 1`] = `
Object {
  "babel": Object {
    "plugins": Array [
      "transform-class-properties",
    ],
    "presets": Array [
      "es2015",
      "react",
    ],
  },
  "path": "/Users/akameco/src/github.com/akameco/read-babelrc-up/fixture/babelrc/.babelrc",
}
`;

exports[`sync 2`] = `
Object {
  "babel": Object {
    "plugins": Array [
      "transform-class-properties",
    ],
  },
  "path": "/Users/akameco/src/github.com/akameco/read-babelrc-up/fixture/pkg/package.json",
}
`;
